### PR TITLE
Stats: update sparkline

### PR DESCRIPTION
**Pull request by jeherve**
@rase- I think it'd be nice if we could merge that old branch you worked on last August. Can you recall why we didn't merge it back then? 

I rebased it, tested it, it seems to work nicely. The only problem I found was that it doesn't play well with other plugins adding items to the admin bar.

e.g. with WP Super Cache:

screen shot 2014-12-15 at 11 24 05 am

and on hover:
screen shot 2014-12-15 at 11 23 08 am

### CODE REVIEW

File: modules/stats.php [hunk 1]
```diff
@@ -68,6 +84,26 @@ function stats_load() {
...
+function jetpack_stats_admin_bar_stats_chart() {
+	$stats = get_transient( 'jetpack_stats_admin_bar_chart' );
+
+	if( false === $stats ) {
+		Jetpack::load_xml_rpc_client();
+		$xml = new Jetpack_IXR_Client();
+		$xml->query( 'jetpack.stats.adminBarChart' );
+		if( $xml->isError() ) {
+			error_log( "Error getting Jetpack Admin Bar Stats: " . $xml->get_jetpack_error()->get_error_message() );
+			echo array();
```
Comment (dereksmart) [inline modules/stats.php]: Why echo an empty array?

File: modules/stats.php [hunk 2]
```diff
@@ -437,210 +474,380 @@ function stats_reports_page() {
...
-
-		stats_set_options( $options );
-		stats_update_blog();
-		Jetpack::state( 'message', 'module_configured' );
-		wp_safe_redirect( Jetpack::module_configuration_url( 'stats' ) );
-		exit;
+	function stats_convert_post_title( $matches ) {
+		global $stats_posts;
+		$post_id = $matches[1];
+		if ( isset( $stats_posts[$post_id] ) )
```
Comment (dereksmart) [inline modules/stats.php]: Whitespace in brackets, and would like to see braces used

File: modules/stats.php [hunk 3]
```diff
@@ -437,210 +474,380 @@ function stats_reports_page() {
...
-		.stats-smiley {
-			vertical-align: 1px;
+	function stats_configuration_load() {
+		if ( isset( $_POST['action'] ) && $_POST['action'] == 'save_options' && $_POST['_wpnonce'] == wp_create_nonce( 'stats' ) ) {
+			$options = stats_get_options();
+			$options['admin_bar']  = isset( $_POST['admin_bar']  ) && $_POST['admin_bar'];
+			$options['hide_smile'] = isset( $_POST['hide_smile'] ) && $_POST['hide_smile'];
+
+			$options['roles'] = array( 'administrator' );
+			foreach ( get_editable_roles() as $role => $details )
```
Comment (dereksmart) [inline modules/stats.php]: missing braces in foreach and if statements

**Comment by samhotchkiss** [comment]
@rase- - can you update us on this?  Also, in addition to Derek's comments, it also needs a rebase

**Comment by dereksmart** [comment]
punted to 3.7 unless someone steps in quick

**Comment by samhotchkiss** [comment]
Closing this for now, we'll probably re-do from scratch down the road, or we can always re-open...

**Pull request closed without merging**